backend/src/main/java/com/redhat/cloud/notifications/db/repositories/ApplicationRepository.java

```diff
@@ -121,6 +121,25 @@ public Application getApplication(String bundleName, String applicationName) {
         }
     }
 
+    /**
+     * Returns applications that have at least 1 forced email notification in a given bundle
+     */
+    public List<Application> getApplicationsWithForcedEmail(UUID bundleId, String orgId) {
+        String query = "SELECT a FROM Application a " +
+                "JOIN a.eventTypes et " +
+                "JOIN et.behaviors etb " +
+                "JOIN etb.behaviorGroup.actions action " +
+                "WHERE a.bundle.id = :bundleId AND (etb.behaviorGroup.orgId is NULL OR etb.behaviorGroup.orgId = :orgId) " +
+                "AND action.id.endpointId IN (" +
+                    "SELECT e.id FROM Endpoint e JOIN EmailSubscriptionProperties props ON e.id = props.id  " +
+                    "WHERE (e.orgId is NULL OR e.orgId = :orgId) AND props.ignorePreferences = true" +
+                ")";
```

Comment:
@gwenneg Wondering if you can think of a simpler query

I'm trying to fetch all the applications that have a forced email action:

Application -> Event Types -> Behavior groups -> Endpoint -> EmailSubscription.forced = true

Reply:
WDYT about this?
```java
        String query = "SELECT a FROM Application a " +
                "JOIN a.eventTypes et " +
                "JOIN et.behaviors etb " +
                "JOIN etb.behaviorGroup.actions action " +
                "WHERE a.bundle.id = :bundleId AND (etb.behaviorGroup.orgId is NULL OR etb.behaviorGroup.orgId = :orgId) " +
                "AND EXISTS (" +
                    "SELECT 1 FROM EmailSubscriptionProperties props  " +
                    "WHERE action.id.endpointId = props.id AND props.ignorePreferences = true" +
                ")";
```

Reply:
Since there's already a condition on the `orgId` for the behavior groups, I think it's safe to skip the `orgId` condition for the endpoints linked to the behavior groups.

Reply:
Looks good! Applied the change, thanks!